connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -344,7 +344,9 @@ class SparkSession private[sql] (
   // scalastyle:on
 
   def newSession(): SparkSession = {
-    throw new UnsupportedOperationException("newSession is not supported")
+    val newClient: SparkConnectClient =
```

Comment:
Small nit. We could also implement `SparkConnectClient.copy()`. That way we don't have to expose as much of the client.

Reply:
This is a very good idea.